lib/phoenix_live_view.ex

```diff
@@ -959,40 +959,60 @@ defmodule Phoenix.LiveView do
   ## Options
 
     * `:to` - the path to redirect to. It must always be a local path
+    * `:status` - the HTTP status code to use for the redirect. Defaults to 302.
     * `:external` - an external path to redirect to. Either a string
       or `{scheme, url}` to redirect to a custom scheme
 
   ## Examples
 
       {:noreply, redirect(socket, to: "/")}
+      {:noreply, redirect(socket, to: "/", status: 301)}
       {:noreply, redirect(socket, external: "https://example.com")}
 
   """
-  def redirect(socket, opts \\ [])
+  def redirect(socket, opts \\ []) do
+    cond do
+      Keyword.has_key?(opts, :to) ->
+        do_internal_redirect(socket, Keyword.fetch!(opts, :to), Keyword.get(opts, :status))
```

Comment:
I would just always pass it explicitly `Map.get(opts, :status, 302)` and have the rest of the code assume it is there. It should be much simpler downstream!

Reply:
Thanks, @josevalim ! I updated the PR; I initially didn't do this because I wanted to keep the old behavior identical; if the caller doesn't mention a status don't set a status. If you are comfortable w/ the tests that had to update to support always putting a status in there, I do agree it's cleaner!